### PR TITLE
Move parameter attributes into ItemCondition and ItemEffect class

### PR DESCRIPTION
**Pull request by JaskRendix**
PR addresses the removal of namedtuple from item (effects and conditions), now the attributes are on the class.
Inspired by #1385 

Addition:
- added **hp** parameter to **revive** as well as 1 in the JSON

Everything works, there is the message when you can use an item or not, you can use the capture device and capture monsters, you can use potion on monster, etc.

### CODE REVIEW

File: tuxemon/item/conditions/shape.py [hunk 1]
```diff
@@ -48,14 +42,18 @@ class ShapeCondition(ItemCondition[ShapeConditionParameters]):
     """
 
     name = "shape"
-    param_class = ShapeConditionParameters
+    shape1: str
+    shape2: Union[str, None]
+    shape3: Union[str, None]
```
Comment (kerizane) [inline tuxemon/item/conditions/shape.py]: I think these need to be defaulted to None. Can you test?

Comment (JaskRendix) [reply tuxemon/item/conditions/shape.py]: You're absolutely right

**Comment by kerizane** [comment]
In my PR I have just added "test_actions" to validate all existing usages. Is this something you can do in this PR?

**Comment by JaskRendix** [comment]
That's a pretty good script, I'm not so skilled in scripts, but you mean iterating all the files inside the folder /mods/db/item with ".json" inside the "effects or conditions" field?

File: tuxemon/item/conditions/__init__.py [hunk 2]
```diff
@@ -1,26 +0,0 @@
-#
```
Comment (kerizane) [inline tuxemon/item/conditions/__init__.py]: Sorry, I would rather you leave these in for now, until I have confirmation that it's ok to remove them.

If you are creating a file from scratch then feel free to leave it out.

Comment (JaskRendix) [reply tuxemon/item/conditions/__init__.py]: No problem, I'm going to reverse.

**Review by kerizane — APPROVED** [review]

**Pull request merged**